tools/firmware-utils/src/mktplinkfw2.c

```diff
@@ -156,6 +156,12 @@ static struct flash_layout layouts[] = {
 		.kernel_ep	= 0x80000000,
 		.rootfs_ofs	= 0x140000,
 	}, {
+		.id		= "8MLmtk",
+		.fw_max_len	= 0x7b0000,
```

Comment:
At it's current form, sysupgrade does not appear to be able to keep the settings between each flash. the overlay partition keeps getting re-built after sysupgrade restart.